### PR TITLE
Application source edit, fix form behavior

### PR DESCRIPTION
**Pull request by torchiaf**
### Summary
Fixes https://github.com/epinio/ui/issues/216

### Areas or cases that should be tested

Applications -> `GitPicker`

### CODE REVIEW

File: pkg/epinio/components/application/AppSource.vue [hunk 1]
```diff
@@ -99,6 +99,7 @@ export default Vue.extend<Data, any, any, any>({
       },
 
       git: {
+        type:          sourceType,
```
Comment (richard-cox) [inline pkg/epinio/components/application/AppSource.vue:102]: this gets persisted, so duplicates the type in the env var. what does it fix here?

File: pkg/epinio/detail/applications.vue [hunk 2]
```diff
@@ -215,7 +215,7 @@ export default Vue.extend<Data, any, any, any>({
 
     commitPosition() {
       if (!this.preparedCommits.length && !this.gitDeployment.deployedCommit) {
-        return;
+        return null;
```
Comment (richard-cox) [inline pkg/epinio/detail/applications.vue:218]: i'm not sure these changes do anything, commitPosition is only used in a single place. the prop it's passed to converts the undefined to defaults, and within the component it's just `v-if="commitPosition"` anyway?

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: pkg/epinio/components/application/AppSource.vue (hunk 1 above)
Comment (torchiaf) [reply pkg/epinio/components/application/AppSource.vue:102]: This is the type from data source and it's compared to the one from LabeledInput:
```
    if (this.value.type && this.value.type !== this.ghType) {
      return;
    }
```

File: pkg/epinio/detail/applications.vue (hunk 2 above)
Comment (torchiaf) [reply pkg/epinio/detail/applications.vue:218]: Yes, we can remove this fix sorry.

**Comment by torchiaf** [comment]
Thanks for the review @richard-cox . I made some fixes to address your CRs.
Last [commit](https://github.com/rancher/dashboard/pull/8672/commits/0a3846963cc57f73416acd8edc18a4f4a0d8ebd1) fixes a case when GitLab source is selected and the 'Update Source' button remains active , please see the description of the issue, it is updated with this last case.

**Comment by richard-cox** [comment]
@torchiaf Worth re-visiting this one once the other PRs merge

**Comment by torchiaf** [comment]
> @torchiaf Worth re-visiting this one once the other PRs merge

Done.

File: shell/components/form/GitPicker.vue [hunk 3]
```diff
@@ -73,58 +73,14 @@ export default Vue.extend<Data, any, any, any>({
...
-          .then(() => {
-            if (this.branches.length && !this.hasError.branch) {
-              this.selectedBranch = this.value.branch;
-
-              return this.fetchCommits();
-            }
-          })
-          .then(() => {
-            if (this.commits.length && this.commits.find((c: commit) => c.sha === this.value.commit || c.id === this.value.commit)) {
-              this.selectedCommit = { sha: this.value.commit };
```
Comment (richard-cox) [inline shell/components/form/GitPicker.vue:115]: This line hasn't made it over to the new location. It means the top commit is always selected regardless of whether it's the correct one (see `richard/rc-github` app. `chatty-app` commit deployed, but `Create README.md Added README` selected)

**Review by richard-cox — CHANGES_REQUESTED** [review]
One bug, otherwise looking good

**Pull request merged**